src/python/zigopt/authentication/result.py

```diff
@@ -1,27 +1,22 @@
 # Copyright © 2022 Intel Corporation
 #
 # SPDX-License-Identifier: Apache License 2.0
-def authentication_result(
-  token=None,
-  client=None,
-  user=None,
-  permission=None,
-  membership=None,
-  authenticated_from_email_link=None,
-  authenticated_public_cert=None,
-  authorization_response=None,
-  session_expiration=None,
-  organization=None,
-):
-  return {
-    "token": token,
-    "client": client,
-    "user": user,
-    "permission": permission,
-    "membership": membership,
-    "authenticated_from_email_link": authenticated_from_email_link,
-    "authenticated_public_cert": authenticated_public_cert,
-    "authorization_response": authorization_response,
-    "session_expiration": session_expiration,
-    "organization": organization,
-  }
+from typing import TypedDict
+
+from zigopt.client.model import Client
+from zigopt.membership.model import Membership
+from zigopt.organization.model import Organization
+from zigopt.permission.model import Permission
+from zigopt.token.model import Token
+from zigopt.user.model import User
+
+
+class authentication_result(TypedDict, total=False):
```

Comment:
we can make this `AuthenticationResult` if we're gonna use a class / pass around objects. so, instead of "calling a func that returns a dict", we're gonna "call a class constructor to get the obj, which will behave like a dict, plus type-checking"? Seems like it should work, just confirming